configure.ac

```diff
@@ -717,13 +726,20 @@ AC_SUBST(OLA_CLIENT_LIBS)
 
 # Extra tools
 #####################################################
-AM_CONDITIONAL([BUILD_JA_RULE],
-               [test "x$enable_ja_rule" = xyes && test "x$have_libusb" = xyes \
-                && test "x$have_libusb_hotplug_api" = xyes])
+AS_IF([test "x$enable_ja_rule" = xyes && test "x$enable_usbdmx" != xyes],
```

Comment:
Hmm I don't think it's that simple. Outputting the error message only makes sense if $enable_ja_rule is exactly "yes" (forced enabled by the user). Assuming Ja Rule has an auto mode (enabled if possible, otherwise disabled), if the check is "$enable_ja_rule != no && $enable_usbdmx == no" as you suggest, then this will fail:

    ./configure --enable-usbdmx=no

whereas we would just want to silently disable Ja Rule.

If we want to have an auto mode, I think we will have to add an "else if" clause to the existing AS_IF. In pseudo-code, it would look like:

    if $enable_ja_rule == yes && $enable_usbdmx != yes:
        error("Ja Rule needs libusb...")
    elif $enable_ja_rule == auto $$ $enable_usbdmx != yes:
        $enable_ja_rule = no

This would be repeated for each of Ja Rule's dependencies. Note that if $enable_usbdmx was originally set to auto, at this point it would have been forced to either "yes" or "no" by the code in PLUGIN_SUPPORT. Similarly, after checking for all Ja Rule's dependencies, we should force enable_ja_rule to either "yes" or "no", so that if some other code depends on it, it can check whether Ja Rule will actually be enabled or not.

Does this make sense?